mono/mini/llvm-jit.cpp

```diff
@@ -284,26 +285,34 @@ class MonoLLVMJIT {
 		initializeScalarOpts(registry);
 		initializeInstCombine(registry);
 		initializeTarget(registry);
+		linkCoreCLRGC(); // Mono uses built-in "coreclr" GCStrategy
 
+		// FIXME: find optimal mono specific order of passes
+		// see https://llvm.org/docs/Frontend/PerformanceTips.html#pass-ordering
+		// the following order is based on a stripped version of "OPT -O2"
+		const char *default_opts = " -simplifycfg -sroa -lower-expect -instcombine -licm -simplifycfg -lcssa -indvars -loop-deletion -gvn -memcpyopt -sccp -bdce -instcombine -dse -simplifycfg";
 		const char *opts = g_getenv ("MONO_LLVM_OPT");
-		if (opts == NULL) {
-			// FIXME: find optimal mono specific order of passes
-			// see https://llvm.org/docs/Frontend/PerformanceTips.html#pass-ordering
-			// the following order is based on a stripped version of "OPT -O2"
-			opts = " -simplifycfg -sroa -lower-expect -instcombine -licm -simplifycfg -lcssa -indvars -loop-deletion -gvn -memcpyopt -sccp -bdce -instcombine -dse -simplifycfg";
-		}
+		if (opts == NULL)
+			opts = default_opts;
+		else if (opts[0] == '+') // Append passes to the default order if starts with '+', overwrite otherwise
+			opts = g_strdup_printf ("%s %s", default_opts, opts + 1);
+		else if (opts[0] != ' ') // pass order has to start with a leading whitespace
+			opts = g_strdup_printf (" %s", opts);
```

Comment:
Unrelated change, just improved `MONO_LLVM_OPT` a bit (it's now possible to append passes to the default order instead of completely overwrite them.